go/vt/vttablet/tabletserver/tx_engine.go

```diff
@@ -94,6 +96,11 @@ type TxEngine struct {
 	dxNotify     func()
 }
 
+const (
+	TwoPCAllowed_SemiSync = iota
+	TwoPCAllowed_TabletControls
+)
+
 // NewTxEngine creates a new TxEngine.
```

Comment:
Let's add one for const `TwoPCAllowed_TypeLen` in the end and use that to initialize `te.twopcAllowed`